starlite/middleware/csrf.py

```diff
@@ -37,6 +38,12 @@ def __init__(
         """
         self.app = app
         self.config = config
+        self.exclude: Optional[Pattern[str]] = None
```

Comment:
This piece of code has been duplicated from `AbstractAuthenticationMiddleware`.

It feels like we can think and generalize skipping in middleware processing but I still need to play with websocket/generic middlewares first before I can come up with good protocol change.

Reply:
It doesn't work with protocols I'm afraid. We can introduce an AbstractBaseMiddleware